### PR TITLE
Show/hide nav bar and toolbar when scrolling

### LINKED ISSUE #14244: Hide/show header and toolbar when scrolling

### PR DESCRIPTION
**Pull request by leandroalonso**
Fixes #14244

Review only after #14283 is merged.

This PR adds the feature to hide/show the navigation bar and the toolbar while the user scrolls.

To achieve a more smooth hiding/showing I used the https://github.com/andreamazz/AMScrollingNavbar library. It's a very small and popular library.

The reason I didn't use `hidesBarsOnSwipe` is that the transition is kinda glitchy. Plus, that would require more custom code to hide the toolbar (and apparently not even Apple uses it).

### To test

**Make sure that you have overridden the new reader Feature Flag and it's not using the older Reader (Me -> App Settings -> Debug -> "Reader content displayed in a webview" should be ON)**

1. Open a post in the Reader
2. Scroll up, down, go back to the post lists, open another one, repeat :)

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift [hunk 1]
```diff
@@ -208,6 +226,46 @@ class ReaderDetailWebviewViewController: UIViewController, ReaderDetailView {
...
+    ///
+    func scrollViewShouldScrollToTop(_ scrollView: UIScrollView) -> Bool {
+        if let navigationController = navigationController as? ScrollingNavigationController {
+            navigationController.showNavbar(animated: true, scrollToTop: true)
+        }
+        return true
+    }
+
+    /// Update scroll view insets to take into account if toolbar is visible or not
+    private func updateScrollInsets(toolbarVisibile: Bool) {
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift:264]:
```suggestion
    private func updateScrollInsets(toolbarVisible: Bool) {
```

File: WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift [hunk 2]
```diff
@@ -208,6 +226,46 @@ class ReaderDetailWebviewViewController: UIViewController, ReaderDetailView {
...
+        let bottomInset: CGFloat = toolbarVisibile ? Constants.toolbarHeight : 0
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift:265]:
```suggestion
        let bottomInset: CGFloat = toolbarVisible ? Constants.toolbarHeight : 0
```

File: WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift [hunk 3]
```diff
@@ -208,6 +226,46 @@ class ReaderDetailWebviewViewController: UIViewController, ReaderDetailView {
...
+    /// Only if VoiceOver is not active
+    ///
+    private func followScrollView() {
+        if isFollowingScrollView,
+            UIAccessibility.isVoiceOverRunning {
+            return
+        }
+
+        if let navigationController = navigationController as? ScrollingNavigationController {
+            navigationController.followScrollView(scrollView, delay: 50.0, followers: [NavigationBarFollower(view: toolbarContainerView, direction: .scrollDown)])
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift:239]: Maybe move the delay to a constant?

File: WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift [hunk 4]
```diff
@@ -208,6 +226,46 @@ class ReaderDetailWebviewViewController: UIViewController, ReaderDetailView {
...
+    private func stopFollowingScrollView() {
+        if let navigationController = navigationController as? ScrollingNavigationController {
+            navigationController.stopFollowingScrollView(showingNavbar: true)
+            isFollowingScrollView = false
+        }
+    }
+
+    /// Show the nav bar when scrolling to top
+    ///
+    func scrollViewShouldScrollToTop(_ scrollView: UIScrollView) -> Bool {
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift:256]: Should this method be grouped in the UIScrollViewDelegate extension below?

File: WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift]: Done in 30fe93e6b3173d9ca702d534d42a65ddc86e5b9b

File: WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift]: Done in 30fe93e6b3173d9ca702d534d42a65ddc86e5b9b

File: WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift (hunk 3 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift]: Done in f1e5981a63ff162a065197703de3e41016a008c7

File: WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift (hunk 4 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift]: Done in bb12fd1b9e109fa7ce742832400510367ef8e00e

**Pull request merged**